### PR TITLE
fix: DEV-2931: Fix hiding NER regions with new outliner

### PR DESCRIPTION
**Pull request by nicholasrq**

### CODE REVIEW

File: src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx [hunk 1]
```diff
@@ -309,11 +309,20 @@ const RegionControls: FC<RegionControlsProps> = observer(({
   const onToggleHidden = useCallback(() => {
     if (type.match('region')) {
       entity.toggleHidden();
+      console.log({ entity });
```
Comment (bmartel) [inline src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx:312]: forgot console.log

**Review by bmartel — CHANGES_REQUESTED** [review]

File: src/mixins/SpanText.js [hunk 2]
```diff
@@ -177,6 +177,7 @@ export default types
     toggleHidden(e) {
       self.hidden = !self.hidden;
       self.setHighlight(self.highlighted);
+      console.log(self.hidden);
```
Comment (bmartel) [inline src/mixins/SpanText.js:180]: here too

**Review by bmartel — APPROVED** [review]

File: src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx [hunk 3]
```diff
@@ -314,6 +314,14 @@ const RegionControls: FC<RegionControlsProps> = observer(({
     }
   }, [item, item?.toggleHidden, hidden]);
 
+  const onToggleCollapsed = useCallback((e: MouseEvent) => {
+    onToggleCollapsed(e);
+  }, [toggleCollapsed]);
+
+  const onToggleLocked = useCallback(() => {
+    item.setLocked(!item.locked);
+  }, [item]);
```
Comment (hlomzik) [inline src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx:323]: maybe `item.locked` as dependency? or even `item.setLocked(someFn)` to always use current value.

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx (hunk 3 above)
Comment (nicholasrq) [reply src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx]: Fixed

File: src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx [hunk 4]
```diff
@@ -314,6 +314,14 @@ const RegionControls: FC<RegionControlsProps> = observer(({
     }
   }, [item, item?.toggleHidden, hidden]);
 
+  const onToggleCollapsed = useCallback((e: MouseEvent) => {
+    onToggleCollapsed(e);
```
[the remark this answers is not in the document]
Comment (nicholasrq) [reply src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx]: Fixed

**Review by hlomzik — APPROVED** [review]

**Pull request merged**